### PR TITLE
Iter5

### PR DESCRIPTION
**Pull request by romanyakovlev**

### CODE REVIEW

File: cmd/shortener/config.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"flag"
+	"log"
+
+	"github.com/caarlos0/env/v6"
+)
+
+var flagAAddr string
```
Comment (arttet) [inline cmd/shortener/config.go]: CHANGE REQUEST: please don't use global variables

File: cmd/shortener/config.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package main
```
Comment (arttet) [inline cmd/shortener/config.go]: CHANGE REQUEST: please update Go layouts
https://github.com/golang-standards/project-layout

File: cmd/shortener/main.go [hunk 3]
```diff
@@ -1,3 +1,7 @@
 package main
 
-func main() {}
+func main() {
+	if err := run(); err != nil {
+		panic(err)
```
Comment (arttet) [inline cmd/shortener/main.go]: CHANGE REQUEST: please don't use `panic`

File: cmd/shortener/server.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+)
+
+func saveURL(shortener URLShortener, w http.ResponseWriter, r *http.Request) {
```
Comment (arttet) [inline cmd/shortener/server.go]: CHANGE REQUEST: it should be a method

File: cmd/shortener/server.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	value, ok := shortener.GetURL(shortURL)
+	if ok {
+		w.Header().Set("Location", value)
+		w.WriteHeader(http.StatusTemporaryRedirect)
+	} else {
+		w.WriteHeader(http.StatusBadRequest)
+	}
+}
+
+func URLShortenerRouter(shortener URLShortener) chi.Router {
```
Comment (arttet) [inline cmd/shortener/server.go]:
```suggestion
func Router(shortener URLShortener) chi.Router {
```

File: cmd/shortener/service.go [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
```
Comment (arttet) [inline cmd/shortener/service.go]: CHANGE REQUEST: please format your code

**Review by arttet — CHANGES_REQUESTED** [review]
Please add
- https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
- https://github.com/golangci/golangci-lint

File: cmd/shortener/config.go (hunk 1 above)
Comment (romanyakovlev) [reply cmd/shortener/config.go]: done

File: cmd/shortener/config.go (hunk 2 above)
Comment (romanyakovlev) [reply cmd/shortener/config.go]: done

File: cmd/shortener/main.go (hunk 3 above)
Comment (romanyakovlev) [reply cmd/shortener/main.go]: done

File: cmd/shortener/server.go (hunk 4 above)
Comment (romanyakovlev) [reply cmd/shortener/server.go]: done

File: cmd/shortener/server.go (hunk 5 above)
Comment (romanyakovlev) [reply cmd/shortener/server.go]: done

File: cmd/shortener/service.go (hunk 6 above)
Comment (romanyakovlev) [reply cmd/shortener/service.go]: done

**Review by arttet — APPROVED** [review]

**Pull request merged**